v2/tools/generator/internal/astmodel/property_reference_test.go

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astmodel
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestPropertyReference_IsEmpty_ReturnsExpectedResult(t *testing.T) {
+	pkg := makeTestLocalPackageReference("Demo", "v1")
+	declaringType := MakeTypeName(pkg, "Person")
+
+	cases := []struct {
+		name          string
+		declaringType TypeName
+		property      PropertyName
+		expected      bool
+	}{
+		{"Property name present means not empty", EmptyTypeName, "property", false},
+		{"Declaring type present means not empty", declaringType, "", false},
+		{"Fully populated means not empty", declaringType, "property", false},
+		{"Totally empty not empty", EmptyTypeName, "", true},
+	}
+
+	t.Parallel()
```

Comment:
super minor: Usually I think t.Parallel() is put up at the top.

Reply:
Moved.